### PR TITLE
tests: fix classic ubuntu core transition auth

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Clean up the /root/.snap and /home/test/.snap properly
    
    This test is failing in the nightly execution.
    
    invariant-tool: the following files should not be owned by root
    /home/test/.snap
    /home/test/.snap/auth.json
    invariant-tool: root-files-in-home not-ok
    invariant-tool: crashed-snap-confine ok
    invariant-tool: system is corrupted

### CODE REVIEW

File: tests/nightly/classic-ubuntu-core-transition-auth/task.yaml [hunk 1]
```diff
@@ -34,9 +47,12 @@ execute: |
     snap ack ./ubuntu-core_*.assert
     snap install ./ubuntu-core_*.snap
 
-    mkdir -p /root/.snap/
+    ensure_dir_exists_backup_real /root/.snap
+    ensure_file_exists_backup_real /root/.snap/auth.json
     echo '{}' > /root/.snap/auth.json
-    mkdir -p /home/test/.snap/
+
+    ensure_dir_exists_backup_real /home/test/.snap
```
Comment (zyga) [inline tests/nightly/classic-ubuntu-core-transition-auth/task.yaml:54]: Does this create the file as the test user? If not then this is probably not a realistic scenario.

File: tests/nightly/classic-ubuntu-core-transition-auth/task.yaml [hunk 2]
```diff
@@ -20,9 +20,22 @@ debug: |
     . "$TESTSLIB/changes.sh"
     snap change "$(change_id 'Transition ubuntu-core to core')" || true
 
+restore: |
+    # shellcheck source=tests/lib/files.sh
+    . "$TESTSLIB/files.sh"
+
+    clean_file /root/.snap/auth.json
```
Comment (zyga) [inline tests/nightly/classic-ubuntu-core-transition-auth/task.yaml:27]: Mild preference to have executables over functions for the reasons of avoiding the import, avoiding shared state with the test.

File: tests/nightly/classic-ubuntu-core-transition-auth/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/nightly/classic-ubuntu-core-transition-auth/task.yaml:54]: Agree, test is previously created during the preparation of the suite.

File: tests/nightly/classic-ubuntu-core-transition-auth/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/nightly/classic-ubuntu-core-transition-auth/task.yaml:27]: This is a good suggestion, I'll try to move those functions to executables in a following pr. This change will affect many tests.

**Comment by sergiocazzolato** [comment]
@zyga thanks for the review

**Pull request merged**